tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Grpc/GrpcLegacy/Client/GrpcLegacyClientCommon.cs

```diff
@@ -76,7 +77,7 @@ internal static class GrpcLegacyClientCommon
 
                 if (setSamplingPriority && existingSpanContext?.SamplingPriority is not null)
                 {
-                    span.SetTraceSamplingPriority(existingSpanContext.SamplingPriority.Value);
+                    span.Context.TraceContext.SetSamplingPriority(existingSpanContext.SamplingPriority);
```

Comment:
In other places I've seen `span.Context.TraceContext?.SetSamplingPriority` (the null-conditional on `TraceContext`), but it isn't on this one or the one below on ln 215 - do we know that the `TraceContext` isn't null here? Is it because we are just starting the span right before this?

Reply:
Good observation. If you track where `parentContext` comes from (it's a `ReadOnlySpanContext`) and what happens when we start the span (`parentContext` is not a `SpanContext`, so we start a new trace with a new `TraceContext`), then you can see  `span.Context.TraceContext` won't be null. In fact, I think for new spans, the trace context can never be null.

There is a case where a span's _parent_ is a `SpanContext` that was extracted from upstream headers (distributed tracing), and in that case, that `SpanContext.TraceContext` will be null. Because of this I tend to add the null checks just to be super extra safe. I will add it here for now, but I hope we can refactor how we use `SpanContext` soon to clean up that duality.